src/components/tree-item/tree-item.tsx

```diff
@@ -0,0 +1,291 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Element,
+  Prop,
+  State,
+  Listen,
+  h,
+  getAssetPath,
+  Watch
+} from "@stencil/core";
+@Component({
+  tag: "gxg-tree-item",
+  styleUrl: "tree-item.scss",
+  shadow: true,
+  assetsDirs: ["tree-item-assets"]
+})
+
+// return "./card-assets/new-card.svg";
+export class GxgTreeItem {
+  //PROPS
+  /**
+   * Set this attribute if you want the gxg-treeitem to display a checkbox
+   */
+  @Prop() checkbox = false;
+
+  /**
+   * Set this attribute if you want the gxg-treeitem checkbox to be checked by default
+   */
+  @Prop() checked = false;
+
+  /**
+   * Set this attribute when you are downloading a resource
+   */
+  @Prop() downloading = false;
+
+  /**
+   * Set thhe left side icon from the available Gemini icon set : https://gx-gemini.netlify.app/?path=/story/icons-icons--controls
+   */
+  @Prop() leftIcon: string;
+  /**
+   
+
+  /**
+   * Set thhe right side icon from the available Gemini icon set : https://gx-gemini.netlify.app/?path=/story/icons-icons--controls
+   */
+  @Prop() rightIcon: string;
+  /**
+
+  /**
+   * If this tree-item has a nested tree, set this attribute to make the tree open by default
+   */
+  @Prop() open = false;
+
+  //STATE
+  @Prop() isLeaf: boolean;
```

Comment:
It could be a good idea to initialize this property.

Reply:
Done!.
Why is not a good idea to check if the item has a tree inside, and define isLeaf above that result?